src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs

```diff
@@ -550,7 +550,7 @@ private static unsafe int CreateArrayGCDesc(LowLevelList<bool> bitfield, int ran
             {
                 if (numSeries > 0)
                 {
-                    *ptr-- = (short)((bitfield.Count - last + baseOffset - 2) * IntPtr.Size);
+                    *ptr-- = (short)((i - last) * IntPtr.Size);
```

Comment:
`baseOffset` is not a factor in computing the last "skip" in a GC series element. It should be just  
the end of the element - offset of the last GC pointer.  

For SZ array the `baseOffset` is `2`, so adding `baseOffset - 2` was harmless. 
In MD array it results in incorrect "skip".